src/claim_proofs/claim_proofs.rs

```diff
@@ -43,17 +43,26 @@
 
 use super::pedersen_commitments::PedersenGenerators;
 use curve25519_dalek::{ristretto::RistrettoPoint, scalar::Scalar};
-use lazy_static::lazy_static;
-use schnorrkel::{context::SigningContext, signing_context, Keypair, PublicKey, Signature};
+use merlin::Transcript;
+use rand_chacha::ChaChaRng;
+use rand_core::SeedableRng;
+use schnorrkel::{context::SigningTranscriptWithRng, Keypair, PublicKey, Signature};
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 use sp_std::prelude::*;
 
 /// Signing context.
 const SIGNING_CTX: &[u8] = b"PolymathClaimProofs";
-
-lazy_static! {
-    static ref SIG_CTXT: SigningContext = signing_context(SIGNING_CTX);
+const MESSAGE_LABEL: &[u8] = b"PolymathMessageBytes";
+
+/// Make a signing transcript from the message.
+/// Use ChaCha20 as the PRNG instead of defaulting to the OS RNG. This has to be done
+/// to support platforms that do not provide an OS RNG, e.g. WASM.
+fn make_signing_transcript(message: &[u8]) -> SigningTranscriptWithRng<Transcript, ChaChaRng> {
+    let mut t = Transcript::new(SIGNING_CTX);
+    t.append_message(MESSAGE_LABEL, message);
+    let csprng = ChaChaRng::from_seed([0u8; 32]); // DANGER
```

Comment:
I'm not proud of this seed, let's discuss what we should use in production.

Reply:
Also note that the signer and verifier don't have to use the same RNG. I just used the same RNG for both for simplicity, if the security requirements are different on the prover/signer/investor side than the verifier/blockchain we can change this.

Reply:
@ajivanyan In these 2 places it's mentioned that using a deterministic RNG with Schnorrkel isn't advised, but does it apply to the Claim proofs application of Shnorrkel?

>  However, the choice to make the blinding factors fully deterministic makes fault injection attacks much easier, which has been used with some success on Ed25519.

from https://merlin.cool/transcript/rng.html.
and:
/// There 

> are tricks like `attach_rng(t,ChaChaRng::from_seed([0u8; 32]))`
> /// for deterministic tests.  We warn against doing this in production
> /// however because, although such derandomization produces secure Schnorr
> /// signatures, we do implement protocols here like multi-signatures which
> /// likely become vulnerable when derandomized.
from https://github.com/w3f/schnorrkel/blob/cfdbe9ae865a4d3ffa2566d896d4dbedf5107028/src/context.rs#L380.